wallet/wallet/hd_wallet.go

```diff
@@ -411,6 +413,75 @@ func (w *HDWallet) deriveKeyNodeV2(nextIndex uint32) (*slip10.Node, error) {
 	return keyNode, nil
 }
 
+func (w *HDWallet) removeTaintedKeyFromRestrictedKeys(taintedKeyPair HDKeyPair) {
+	allKeysAreTainted := w.areAllKeysTainted()
+
+	for hostname, permissions := range w.permissions {
+		if !permissions.PublicKeys.Enabled() {
+			continue
+		}
+
+		if !permissions.PublicKeys.HasRestrictedKeys() {
```

Comment:
I think this could be folded into:

```
if !permissions.PublicKeys.HasRestrictedKeys() && allKeysAreTainted {
      permissions.PublicKeys = NoPublicKeysPermission()
      w.permissions[hostname] = permissions
}
```

the continue will take place anyways because the slice is empty so `taintedKeyIdx` will equal -1 and will result in continue.

Reply:
sure, but I think the intention is clearer as it is right now. Not adding the `continue` right after might let think there is something that could be done whereas it should actually be a terminal action.

Reply:
And it also prevent further evaluation. That could be an extra point.